model/src/main/kotlin/Identifier.kt

```diff
@@ -147,22 +147,23 @@ data class Identifier(
         }
 
     /**
-     * Map a package manager type as to a package url using the package type.
-     * Returns null when package manager cannot be mapped to a package type.
+     * Map a package manager type to the String representation of the respective [PurlType].
+     * Falls back to the lower case package manager type if the [PurlType] cannot be determined unambiguously.
+     *
+     * E.g. PIP to [PurlType.PYPI] or Gradle to [PurlType.MAVEN].
      */
     fun getPurlType() = when (type.toLowerCase()) {
-        "bower" -> PurlType.BOWER
-        "bundler" -> PurlType.GEM
-        "cargo" -> PurlType.CARGO
-        "carthage", "pub", "spdx", "stack" -> null
-        "composer" -> PurlType.COMPOSER
-        "conan" -> PurlType.CONAN
-        "dep", "glide", "godep", "gomod" -> PurlType.GOLANG
-        "dotnet", "nuget" -> PurlType.NUGET
-        "gradle", "maven", "sbt" -> PurlType.MAVEN
-        "npm", "yarn" -> PurlType.NPM
-        "pip", "pipenv" -> PurlType.PYPI
-        else -> null
+        "bower" -> PurlType.BOWER.toString()
+        "bundler" -> PurlType.GEM.toString()
+        "cargo" -> PurlType.CARGO.toString()
+        "composer" -> PurlType.COMPOSER.toString()
+        "conan" -> PurlType.CONAN.toString()
+        "dep", "glide", "godep", "gomod" -> PurlType.GOLANG.toString()
+        "dotnet", "nuget" -> PurlType.NUGET.toString()
+        "gradle", "maven", "sbt" -> PurlType.MAVEN.toString()
+        "npm", "yarn" -> PurlType.NPM.toString()
+        "pip", "pipenv" -> PurlType.PYPI.toString()
+        else -> type.toLowerCase()
```

Comment:
Let's fix the indentation at the same time and write this as

    fun getPurlType() =
        when (val lowerType = type.toLowerCase()) {
            "bower" -> PurlType.BOWER
            "bundler" -> PurlType.GEM
            "cargo" -> PurlType.CARGO
            "composer" -> PurlType.COMPOSER
            "conan" -> PurlType.CONAN
            "dep", "glide", "godep", "gomod" -> PurlType.GOLANG
            "dotnet", "nuget" -> PurlType.NUGET
            "gradle", "maven", "sbt" -> PurlType.MAVEN
            "npm", "yarn" -> PurlType.NPM
            "pip", "pipenv" -> PurlType.PYPI
            else -> lowerType
        }.toString()

for a bit less code duplication.